tools/RAiDER/statsPlot.py

```diff
@@ -465,21 +464,30 @@ def __init__(self, filearg, col_name, workdir='./', bbox=None, spacing=1, timein
 
     def _getExtent(self):  #dataset, spacing=1, userbbox=None
         """ Get the bbox, spacing in deg (by default 1deg), optionally pass user-specified bbox. Output array in WESN degrees """
-        extent = [np.floor(min(self.df['Lon'])-(self.spacing/2)), np.ceil(max(self.df['Lon'])+(self.spacing/2)),
-                  np.floor(min(self.df['Lat'])-(self.spacing/2)), np.ceil(max(self.df['Lat'])+(self.spacing/2))]
+        extent = [np.floor(min(self.df['Lon'])), np.ceil(max(self.df['Lon'])),
+                  np.floor(min(self.df['Lat'])), np.ceil(max(self.df['Lat']))]
         if self.bbox is not None:
             dfextents_poly = Polygon(np.column_stack((np.array([extent[0], extent[0], extent[1], extent[1], extent[0]]),
                                                       np.array([extent[2], extent[3], extent[3], extent[2], extent[2]]))))
             userbbox_poly = Polygon(np.column_stack((np.array([self.bbox[2], self.bbox[3], self.bbox[3], self.bbox[2], self.bbox[2]]),
                                                      np.array([self.bbox[0], self.bbox[0], self.bbox[1], self.bbox[1], self.bbox[0]]))))
             if userbbox_poly.intersects(dfextents_poly):
-                extent = [np.floor(self.bbox[2]-(self.spacing/2)), np.ceil(self.bbox[-1]+(self.spacing/2)),
-                          np.floor(self.bbox[0]-(self.spacing/2)), np.ceil(self.bbox[1]+(self.spacing/2))]
+                extent = [np.floor(self.bbox[2]), np.ceil(self.bbox[-1]),np.floor(self.bbox[0]), np.ceil(self.bbox[1])]
             else:
-                print(
-                    "WARNING: User-specified bounds do not overlap with dataset bounds, proceeding with the latter.")
+                print("WARNING: User-specified bounds do not overlap with dataset bounds, proceeding with the latter.")
             del dfextents_poly, userbbox_poly
 
+        # ensure that extents do not exceed -180/180 lon and -90/90 lat
```

Comment:
Should we be changing extents or raising exception? Seems that if points are requested outside the bounds might be a bigger problem.

Reply:
@jlmaurer  This is not just a means to dummy-proof it for user error, but also deal with rounding/precision errors which give an invalid value like -91 lat or 182 lon. Would it be better to raise an exception, or just smooth out such errors under the hood?

Reply:
Personally I think we should raise an exception and then handle the exceptions externally. If @Askahloic or @dbekaert have other thoughts it's not a strong opinion

Reply:
*@Askaholic

Reply:
Maybe split the difference. If the user passes bad bounds in then raise an exception, but if you are computing the bounds then fixup the rounding errors.

Reply:
I agree with @Askaholic, went ahead and now raise an exception for cases where the user-specified bounds are bad.